odev2.py

```diff
@@ -0,0 +1,26 @@
+class Animal:
+    def __init__(self, kind, name):
+        self.kind = kind 
+        self.name = name
+
+class Cat(Animal):
+    def meow(self):
+        print("meow meow !")
+
+
+class Cow(Animal):
+    def moo(self):
+        print("mooo !")
+
+
+class Dog(Animal):
+    def hav(self):
+        print("hav hav !")
+
+c = Cat("animal", "cat")
+c.meow()
+cw = Cow("animal", "cow")
+cw.moo()
+d = Dog("animal", "dog")
+d.hav()
+
```

Comment:
kind diye bir parametreye kesinlikle ihtiyacimiz yok, cunku Animal class'indan inherit alan diger classlar zaten turunu belli ediyor.

Ayrica Yazilim Dunyasinda Adapter Pattern olarak adlandirdigimiz bir teknik var, bu teknige gore gelistirme yaparsak soyle yapabiliriz;
```suggestion
class Animal:
    def __init__(self, name):
        self.name = name
    
    def sound(self):
        raise NotImplementedError()

class Cat(Animal):
    def sound(self):
        print("meow meow !")


class Cow(Animal):
    def sound(self):
        print("mooo !")


class Dog(Animal):
    def sound(self):
        print("hav hav !")

```

`NotImplementedError` -> Bir sinif eger baska siniflar tarafindan inherit alacaksa ve herhangi bir metodu miras alan sinif tarafindan kullanilmasi gerekiyorsa, ilgili metoduna boyle bir exception eklenir. ornegin boylece, koyun diye bir hayvan tanimlayip `sound` metodunu implemente etmezsek, bu hata firlatilacaktir. bu da developer'a o metodu implemente etmesi gerektigini hatirlatir.